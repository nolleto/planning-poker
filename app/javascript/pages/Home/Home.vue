<template>
  <div>
    <h1>Home page</h1>

    <button @click="onClick">Click here for something</button>
  </div>
</template>

<script>
  export default {
    name: 'Home',

    methods: {
      onClick() {
        alert('something')
      }
    }
  };
</script>
