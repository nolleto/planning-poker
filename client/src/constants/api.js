const API = '/api/v1'

export const USERS = `${API}/auth/users`
export const SESSIONS = `${API}/auth/sessions`
export const TEST = `${API}/test`
