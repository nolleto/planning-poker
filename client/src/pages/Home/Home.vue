<template>
  <div>
    <h1>Home page</h1>

    <button
      :disabled="!ui.btnEnable"
      @click="onClick"
    >
      Click here for something
    </button>
  </div>
</template>

<script>
import { get, post, auth } from '../../services/http'

export default {
  name: 'Home',

  data() {
    return {
      ui: {
        btnEnable: true
      }
    }
  },

  methods: {
    onClick () {
      this.ui.btnEnable = false
      get('/api/v1/test')
        .then(({ data }) => alert(data.status))
        .catch((error) => alert(error))
        .finally(() => this.ui.btnEnable = true)
      // post('/api/auth/users', {
      //   email: 'admin@admin.com',
      //   password: 123456
      // })
      // auth()
    }
  }
}
</script>
