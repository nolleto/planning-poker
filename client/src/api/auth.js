import JWT from 'jwt-client'
import { post } from '../services/http'
import { AUTH } from '../constants/api'

const saveToken = ({ data }) => {
  const { authToken } = data
  const session = JWT.read(authToken)

  JWT.keep(session)

  return data
}

export const signIn = (login, password) => {
  return post(AUTH, { login, password })
    .then(saveToken)
}
export const signUp = data => {
  return post(`${AUTH}/sign_up`, data)
    .then(saveToken)
}

export const signOut = async () => {
  JWT.forget()
}
