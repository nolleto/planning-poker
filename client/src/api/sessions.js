import JWT from 'jwt-client'
import http from '../services/http'
import { SESSIONS } from '../constants/api'

const saveToken = ({ data }) => {
  const { authToken } = data
  const session = JWT.read(authToken)

  JWT.keep(session)

  return data
}

export const signIn = (login, password) => {
  return http.post(SESSIONS, { login, password })
    .then(saveToken)
}

export const signOut = () => {
  return http.delete(SESSIONS)
    .then(() => {
      JWT.forget()
    })
}
