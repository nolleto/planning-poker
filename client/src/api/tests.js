import { TEST } from '../constants/api'
import http from '../services/http'

export const test = () => http.get(TEST)
