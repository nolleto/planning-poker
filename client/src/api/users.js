import JWT from 'jwt-client'
import http from '../services/http'
import { USERS } from '../constants/api'

const saveToken = ({ data }) => {
  const { authToken } = data
  const session = JWT.read(authToken)

  JWT.keep(session)

  return data
}

export const create = data => {
  return http.post(USERS, data)
    .then(saveToken)
}
