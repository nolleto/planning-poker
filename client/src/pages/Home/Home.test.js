import * as testApi from '../../api/tests'
import Home from './index'
import { shallowMount } from '@vue/test-utils'

const shallowComponent = () => {
  return shallowMount(Home)
}

describe('Home component', () => {
  let wrapper

  describe('Snapshots', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    it('matches', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Event Listeners', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    describe('When clicked in "Click here for something" button', () => {
      let testApi

      beforeEach(() => {
        testApi = jest.fn()
        wrapper.setMethods({ testApi })
        wrapper.find('button').trigger('click')
      })

      it('calls #testApi', () => {
        expect(testApi).toHaveBeenCalled()
      })
    })
  })

  describe('Methods', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    describe('#testApi', () => {
      beforeEach(() => {
        window.alert = jest.fn()
        testApi.test = jest.fn(async () => {})
        wrapper.vm.testApi()
      })

      it('calls #test method from test API', () => {
        expect(testApi.test).toHaveBeenCalled()
      })
    })
  })
})
