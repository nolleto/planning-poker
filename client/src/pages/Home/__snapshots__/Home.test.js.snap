// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Snapshots matches 1`] = `
<div>
  <h1>
    Home page
  </h1>
   
  <button>
    
    Click here for test API
  
  </button>
</div>
`;
